enhancements/single-node/single-node-openshift-with-workers.md

```diff
@@ -0,0 +1,445 @@
+
+- Make sure worker node ignition files are generated even in bootstrap-in-place
+single control-plane node "none"-platform installation.
+
+### User Stories
+
+- As an OpenShift cluster administrator, I want to install a single control-plane
+node cluster with additional worker nodes, so that my cluster will be able to
+handle larger computation demands.
+
+- As an OpenShift cluster administrator, I want to add worker nodes to my
+existing single control-plane node cluster, so that it'll be able to meet
+growing computation demands.
+
+### API Extensions
+
+Introduce a new topology field in the Ingress config CR
+(`config.openshift.io/v1/ingresses`) called `IngressPlacement`.
+
+### Implementation Details/Notes/Constraints
+
+This new field will have one of these values - `ControlPlane` or `Workers`. 
+
+In addition, continue to allow the `.spec.replicas` and `.spec.nodePlacement`
+parameters in `operator.openshift.io/v1/ingresscontrollers` CRs to be omitted,
+but change the defaulting behavior for these fields.
+
+The value of the `IngressPlacement` field will affect the defaulting behavior
+of the `IngressController`'s `.spec.replicas` and `.spec.nodePlacement`
+parameters.  In the absence of an `IngressController` resource created by the
+user/installer, or when the user/installer creates an `IngressController` with
+these two parameters omitted, the Cluster Ingress Operator will choose the
+default values for those parameters based on the value of `IngressPlacement`.
+
+If the value of `IngressPlacement` itself is omitted by the installer, it
+is defaulted to `Workers`. This would be useful to maintain the current behavior
+if the API change is merged before the installer change to set this field, or if
+we want this newer OCP version to be backwards-compatible with older installers.
+
+When the value of `IngressPlacement` is `Workers`, the defaulting behavior of
+`.spec.replicas` and `.spec.nodePlacement` will be the same as it is today:
+`.spec.replicas` will be chosen according to the value of
+`InfrastructureTopology`, namely `1` when `SingleReplica` or `2` when
+`HighlyAvailable`. `.spec.nodePlacement` will always just be:
+
+```yaml
+nodePlacement:
+  nodeSelector:
+    matchLabels:
+      kubernetes.io/os: linux
+      node-role.kubernetes.io/worker: ''
+```
+
+However, if the value of `IngressPlacement` is `ControlPlane`, the defaulting
+behavior will be different: `.spec.replicas` will be chosen instead according
+to the value of `ControlPlaneTopology`; again, `1` when `SingleReplica` or `2`
+when `HighlyAvailable`. `.spec.nodePlacement` will be always just be:
+
+```yaml
+nodePlacement:
+  nodeSelector:
+    matchLabels:
+      kubernetes.io/os: linux
+      node-role.kubernetes.io/master: ''
+```
+
+(Note that the `kubernetes.io/os: linux` label is mentioned just because it's
+the current behavior, it has no importance in this enhancement)
+
+The installer will detect situations in which it's unlikely the user will want
```

Comment:
@patrickdillon @staebler are you OK with having the installer include this logic?

Reply:
Ongoing discussion at https://github.com/openshift/enhancements/pull/1041#discussion_r817700777